sklearn/tree/_tree.pyx

```diff
@@ -708,8 +708,8 @@ cdef class Tree:
         if self._resize_c(self.capacity) != 0:
             raise MemoryError("resizing tree to %d" % self.capacity)
 
-        cdef Node[::1] node_memory_view = node_ndarray
-        cdef DOUBLE_t[:, :, ::1] value_memory_view = value_ndarray
+        cdef const Node[::1] node_memory_view = node_ndarray
+        cdef const DOUBLE_t[:, :, ::1] value_memory_view = value_ndarray
```

Comment:
Since there is a issue with const memoryviews because of the Windows compiler, we can workaround it by using `PyArray_DATA` directly on the ndarray:

```python
        nodes = memcpy(self.nodes, cnp.PyArray_DATA(node_ndarray),
                       self.capacity * sizeof(Node))
        value = memcpy(self.value, cnp.PyArray_DATA(value_ndarray),
                       self.capacity * self.value_stride * sizeof(double))
```

Reply:
Thank you for the guidance!